third-party/sysdeps/linux/grpc/CMakeLists.txt

```diff
@@ -0,0 +1,147 @@
+if(NOT CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
+  # When included in TheRock, we download sources and set up the sub-project.
+  set(_source_dir "${CMAKE_CURRENT_BINARY_DIR}/source")
+  set(_download_stamp "${_source_dir}/download.stamp")
+
+  therock_subproject_fetch(therock-grpc-sources
+    SOURCE_DIR "${_source_dir}"
+    GIT_REPOSITORY "https://github.com/grpc/grpc.git"
+    GIT_TAG "v1.76.0"
+    GIT_SHALLOW ON
+    GIT_SUBMODULES_RECURSE ON
+    TOUCH "${_download_stamp}"
+  )
```

Comment:
we need to do the clone somehow differently. this takes 10-20 minutes.
The way I've been doing it is

1. shallow clone grpc with submodules into a dir
2. create archive
3. upload somewhere internally
4. build systems fetch that and unarchive

See how gtest does it: https://github.com/ROCm/TheRock/blob/1ce7d2a023399308bc7cfc1a00712724d678b211/third-party/googletest/CMakeLists.txt